### PR TITLE
Add docs and strategies for migration to v2 to v3 of JSX

### PR DESCRIPTION
**Pull request by jchavarri**
My attempt to document v2 to v3 migration approaches.

cc @rickyvetter @yawaramin @alexfedoseev in case you have any ideas to improve it, thanks!

### CODE REVIEW

File: docs/jsx.md [hunk 1]
```diff
@@ -134,4 +136,81 @@ You're effectively passing the array `[| <div />, <div /> |]` to `MyReasonCompon
...
+```reason
+/* In Image.re */
+[@bs.config {jsx: 3}];
+
+[@react.component]
+let make = (~src) => <img src />;
+
+module Jsx2 = {
+  let component = ReasonReact.statelessComponent("Image");
+  let make = (~src, children) =>
```
Comment (jchavarri) [inline docs/jsx.md]: @rickyvetter I'm not sure if `key` has to be added here?

Comment (rickyvetter) [reply docs/jsx.md]: Nope. Not needed!

File: docs/jsx.md [hunk 2]
```diff
@@ -134,4 +136,81 @@ You're effectively passing the array `[| <div />, <div /> |]` to `MyReasonCompon
...
+
+### File level
+
+For larger applications, it might not be possible to migrate all components at once. In these cases, a per-file migration is also possible.
+
+A file can be configured to use version 3 of the transform by adding `[@bs.config {jsx: 3}]` at the top of the file.
+
+The per-file configuration allows to mix, in the same application, components compatible with either of both versions of the JSX transforms. However, the restriction is that all the components used in a file will have to be compatible with the JSX version specified for that file.
+
+For example, if a file contains the following code:
```
Comment (rickyvetter) [inline docs/jsx.md]: Yeah love having concrete examples here.

File: docs/jsx.md (hunk 1 above)
Comment (diligiant) [reply docs/jsx.md]: @jchavarri @rickyvetter rightfully insisted (Discord) that we added ~ to children.

Comment (alexfedoseev) [reply docs/jsx.md]: @diligiant In this case—nope, since it's a wrapper to JSX@2 API: it requires unlabeled last argument.

Comment (diligiant) [reply docs/jsx.md]: @alexfedoseev ok ; don't you think we need a section about these precise "rules" ? Once you know, you know but we might end up with troubles for newbies unless the rls emits a warning/error.

Comment (jchavarri) [reply docs/jsx.md]: @diligiant I added an inline comment in https://github.com/reasonml/reason-react/pull/391/commits/36a4c98858c2077d9421bb22cea3913ae40efbe8 that explains why the label is not needed here. Please let me know if you have any specific suggestions on how to make it better.

Comment (diligiant) [reply docs/jsx.md]: @jchavarri Thank you. It goes back to the difference with React where children is a special prop (3rd param of createElement) ; @rickyvetter was kind enough to highlight this to me.

Likewise, I don't think I've seen the explanation that `=?` inside an interop external is like v2 `[@bs.optional]`, different from normal `=?` (turns the param into `option(a')`).

File: docs/jsx.md [hunk 3]
```diff
@@ -134,4 +136,82 @@ You're effectively passing the array `[| <div />, <div /> |]` to `MyReasonCompon
...
+
+```reason
+/* User.re */
+[@bs.config {jsx: 3}]
```
Comment (baransu) [inline docs/jsx.md]: I think `;` is missing at the end of the line here 🙂

Comment (jchavarri) [reply docs/jsx.md]: @baransu Oh, you're right, thanks! That's why I was getting warnings from the compiler 😅 fixed!

**Comment by nikgraf** [comment]
@jchavarri sooo awesome 👍

**Pull request merged**